### DISCUSSION TITLE
Creating more e2e and integration tests

### DISCUSSION THREAD

**Discussion by Shashwat79802** [wagtail/wagtail#11242] [Development]
I browsed through the repo and noticed that more e2e and integration tests can be created. Can I contribute in this way if possible? I use a tool called keploy for testing and generating datamocks and I feel it'd be helpful to have more tests for the project.

**Comment by laymonage** [comment]
Hello, thank you for your interest in helping out. Yes, I think e2e tests would be very helpful! We've been considering doing more e2e tests but it's not on our priority yet, so we'll definitely appreciate any help we can get. Although I don't think we're going to commit on a new tool for now.

We already have an e2e setup in our CircleCI config, but at the moment it's pretty minimal and it only tests a small subset of the CMS' capabilities. It uses our Django testapp in the codebase to run the tests, which contains many different models for testing the CMS, but does not really reflect a realistic production setup. That said, having more tests with the existing setup would be great. You could start there, the documentation about the test setup is here (it's pretty minimal though): https://docs.wagtail.org/en/stable/contributing/developing.html#integration-tests

Perhaps it's easier to start with bakerydemo: https://github.com/wagtail/bakerydemo, which is somewhat closer to a real project, with clear setup instructions. If we could get e2e tests (e.g. using playwright/puppeteer) that go through the features of the CMS using bakerydemo, and later setting up a CI config to run them automatically every once in a while, that would already be a big improvement.

**Comment by thibaudcolas** [comment]
Hi @Shashwat79802, are you affiliated with Keploy? Do you want to contribute to Wagtail to help us adding more tests with our existing tools, or only if we use the tool you recommend?
